test/create_exercise_test.dart

```diff
@@ -128,5 +129,349 @@ void main() {
         });
       });
     });
+
+    group('templates', () {
+      test('exampleTemplate', () {
+        expect(exampleTemplate('foo bar'), equals('''
+class FooBar {
+
+}
+'''));
+      });
+
+      test('mainTemplate', () {
+        expect(mainTemplate('foo bar'), equals('''
+class FooBar {
+  // Put your code here
+}
+'''));
+      });
+
+      test('testTemplate', () {
+        expect(testTemplate('foo bar'), equals('''
+import 'package:foo_bar/foo_bar.dart';
+import 'package:test/test.dart';
+
+final fooBar = new FooBar();
+
+void main() {
+  group('FooBar', () {
+    test('should work', () {
+      // TODO
+    });
+  });
+}
+'''));
+      });
+
+      test('pubTemplate', () {
+        expect(pubTemplate('foo bar', '1.0.0'), equals('''
+name: 'foo_bar'
+version: 1.0.0
+environment:
+  sdk: '>=2.0.0 <3.0.0'
+dev_dependencies:
+  test: '<2.0.0'
+'''));
+      });
+
+      test('analysisOptionsTemplate', () {
+        expect(analysisOptionsTemplate(), equals('''
+analyzer:
+  strong-mode:
+    implicit-casts: false
+    implicit-dynamic: false
+  errors:
+    unused_element:        error
+    unused_import:         error
+    unused_local_variable: error
+    dead_code:             error
+
+linter:
+  rules:
+    # Error Rules
+    - avoid_relative_lib_imports
+    - avoid_types_as_parameter_names
+    - literal_only_boolean_expressions
+    - no_adjacent_strings_in_list
+    - valid_regexps
+'''));
+      });
+
+      group('testCaseTemplate tests', () {
+        test('simple test for a single case', () {
+          Map<String, dynamic> testCase = <String, dynamic>{
+            "description": "Zero is an Armstrong number",
```

Comment:
Recommend: Replace double quotes with single quotes throught out the file, where appropriate